java/cuvs-java/src/main/java22/com/nvidia/cuvs/internal/CagraIndexImpl.java

```diff
@@ -249,105 +249,109 @@ private IndexReference build() throws Throwable {
    */
   @Override
   public SearchResults search(CagraQuery query) throws Throwable {
-    try (var localArena = Arena.ofConfined()) {
-      checkNotDestroyed();
-      int topK =
-          query.getMapping() != null
-              ? Math.min(query.getMapping().size(), query.getTopK())
-              : query.getTopK();
-      long numQueries = query.getQueryVectors().length;
-      long numBlocks = topK * numQueries;
-      int vectorDimension = numQueries > 0 ? query.getQueryVectors()[0].length : 0;
-      Arena arena = resources.getArena();
+    // Use the shared arena from resources instead of creating a confined arena
+    // to avoid thread safety issues when multiple threads perform searches
```

Comment:
I don't think this is true; if this is a memory issue, using the shared arena will make things worse as it will bound the allocated memory to the lifetime of the resource (they will be freed "too late").

Also, a local confined arena is inherently thread safe, as it is "private" to the thread stack calling this function.

(And also, the localArena is allocated here, but never used -- we use the Arena from `Arena arena = resources.getArena();` a few lines below, so effectively the code is the same -- BTW I think this is a mistake, and I'm fixing it in https://github.com/rapidsai/cuvs/pull/1069)

Reply:
To be clear, I'm not saying there aren't threading issues here, just saying that the localArena is not the culprit.